examples/api-demo-2b-issues-analytics.py

```diff
@@ -41,232 +40,320 @@ def parse_command_line_args():
-                            curIsOnWatchlist=True
-                            self.fixableNewsStoryFound = True
-                        self.callToAction = [curDep['cvssScore'], curDep['title'], curIsOnWatchlist,curDep['packageManager'],curDep['from'][0] , tmpUpgradePath]
-                        self.fixFound=True
-    #Analysis worker
-    def analyzeRepo(self,json_res):
-        for v in json_res['issues']['vulnerabilities']:
-            self.manageDict(self.dictDepIssues[self.iDepIssueCounts], v['package'] + ':' + v['version'])
-            self.vulnIssueTotal +=1
+        print("\n\n***END ANALYSIS***")
+
+    def findCVESInWatchList(self, curDep):  # returnsEmptyIfNotFound
+        curCVEList = curDep.identifiers["CVE"]
+        retValue = ""
+        for curCVE in curCVEList:
+            if self.keyExists(self.cveWatchlist, curCVE):
+                retValue = curCVE
+        return retValue
+
+    def checkCallToAction(self, curDep):
+        tmpUpgradePath = ""
+        curIsOnWatchlist = False
+        if (
+            self.fixableNewsStoryFound == False
+        ):  # check if a news story has been found, if it has, exit, you have your call to action
+            if (
+                curDep.cvssScore > self.callToAction[0]
+            ):  # current issue being looked at is more severe than previous found
+                if (curDep.isUpgradable) or (
+                    curDep.isPatchable
+                ):  # check if a fix is available
+                    if curDep.isUpgradable:  # otherwise no upgrade path
+
+                        if (
+                            curDep.upgradePath[0] != curDep.fromPackages[0]
+                        ):  # not a dependency install issue
+                            tmpUpgradePath = curDep.upgradePath[0]
+                        # else reinstall dependencies
+                        # Bail - this shouldn't be listed
+                    else:
+                        if self.keyExists(curDep, "patches"):
+                            if len(curDep.patches) > 0:  # patches found
+                                tmpUpgradePath = "A custom patch developed/delivered via fix PR or run Snyk Wizard to Apply Snyk Patch!"
+
+                if tmpUpgradePath != "":  # upgrade found
+                    tmpCVE = self.findCVESInWatchList(curDep)
+                    if tmpCVE != "":  # WatchlistItemFound
+                        curIsOnWatchlist = True
+                        self.fixableNewsStoryFound = True
+                    self.callToAction = [
+                        curDep.cvssScore,
+                        curDep.title,
+                        curIsOnWatchlist,
+                        curDep.packageManager,
+                        curDep.fromPackages[0],
+                        tmpUpgradePath,
+                    ]
+                    self.fixFound = True
+
+    # Analysis worker
+    def analyzeRepo(self, issueset):
+        for v in issueset.issues.vulnerabilities:
```

Comment:
I tried running the example, but it breaks at this line.
Traceback (most recent call last):
  File "examples/api-demo-2b-issues-analytics.py", line 358, in <module>
    masterlog.analyzeRepo(issueset)
  File "examples/api-demo-2b-issues-analytics.py", line 305, in analyzeRepo
    for v in issueset.issues.vulnerabilities:
AttributeError: 'IssueSetManager' object has no attribute 'issues'
```suggestion
        for v in issueset.all().issues.vulnerabilities:
```

Reply:
Thanks, fixed.